lightning-c-bindings/README.md

```diff
@@ -0,0 +1,79 @@
+The wrapper crate and C/C++ Headers in this folder are auto-generated from the Rust-Lightning
+source by the c-bindings-gen crate contained in the source tree and
+[cbindgen](https://github.com/eqrion/cbindgen). They are intended to be used as a base for building
+language-specific bindings, and are thus incredibly low-level and may be difficult to work with
+directly.
+
+LDK C Bindings
+===================
+
+The C bindings available at include/lightning.h require inclusion of include/rust_types.h first.
+
+All of the Rust-Lightning types are mapped into C equivalents which take a few forms, namely:
+
+ * Structs are mapped into a simple wrapper containing a pointer to the native Rust-Lightning
+   object and a flag to indicate whether the object is owned or only a reference. Such mappings
+   usually generate a X_free function which must be called to release the allocated resources.
+   Note that calling X_free will do nothing if the underlying pointer is NULL or if the reference
+   flag is set.
+
+   You MUST NOT create such wrapper structs manually, relying instead on constructors which have
+   been mapped from equivalent Rust constructors.
+
+ * Traits are mapped into a concrete struct containing a void pointer for your use and a jump table
+   listing the functions which the trait must implement. The void pointer is free for use as you
+   see fit and passed as the first argument to all function calls, but note that no destructor
```

Comment:
Well, it could be more than that - it doesn't even have to be a pointer. The demo.cpp app just casts file descriptor integers into void* and uses it to store file descriptors. I added a few examples. I dropped the comment about destructor, and added it to the pre-merge todo list at the top of the PR.